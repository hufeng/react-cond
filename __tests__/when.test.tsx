import React from 'react';
import renderer from 'react-test-renderer';
import When from '../src/when';

describe('when test suite', () => {
  it('when true', () => {
    const WhenTrue = () =>
      <When test={true}>
        <div>Hello I am Success</div>
      </When>;

    const component = renderer.create(<WhenTrue />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when false', () => {
    const WhenFalse = () =>
      <When test={false}>
        <div>I am Success</div>
      </When>;

    const component = renderer.create(<WhenFalse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when no children', () => {
    const WhenNoChildren = () => <When test={true} />;

    const component = renderer.create(<WhenNoChildren />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when true has two childrens', () => {
    const WhenElse = () =>
      <When test={true}>
        <div>hello I am Success</div>
        <div>Oop, I am failed</div>
      </When>;

    const component = renderer.create(<WhenElse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when false has two childrens', () => {
    const WhenElse = () =>
      <When test={false}>
        <div>hello I am Success</div>
        <div>Oop, I am failed</div>
      </When>;

    const component = renderer.create(<WhenElse />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when children over 2', () => {
    const WhenMultiple = () =>
      <When test={true}>
        <div>hello I am Success</div>
        <div>Oop, I am failed</div>
        <div>error</div>
      </When>;

    try {
      renderer.create(<WhenMultiple />);
    } catch (err) {
      expect(err).toEqual(
        new Error('When Container can contain a maximum of two childrens')
      );
    }
  });
});
