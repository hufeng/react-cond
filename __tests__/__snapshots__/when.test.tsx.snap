// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when test suite when false 1`] = `null`;

exports[`when test suite when false has two childrens 1`] = `
<div>
  Oop, I am failed
</div>
`;

exports[`when test suite when no children 1`] = `null`;

exports[`when test suite when true 1`] = `
<div>
  Hello I am Success
</div>
`;

exports[`when test suite when true has two childrens 1`] = `
<div>
  hello I am Success
</div>
`;
